### PR TITLE
Auto-detect AttachRequired

### LINKED ISSUE #2: Skip attach should not be a command line argument
It should call the driver to get the driver capabilities to determine the skip attach value

### PR DESCRIPTION
**Pull request by jarrpa**
Resolves #2

### CODE REVIEW

File: cmd/cluster-driver-registrar/main.go [hunk 1]
```diff
@@ -84,17 +79,40 @@ func main() {
...
 	}
 
-	// Get CSI driver name.
-	glog.V(1).Infof("Calling CSI driver to discover driver name.")
+	// Get connection context
 	ctx, cancel := context.WithTimeout(context.Background(), csiTimeout)
 	defer cancel()
+
+	// Get CSI driver name.
+	glog.V(1).Infof("Calling CSI driver to discover driver name.")
```
Comment (davidz627) [inline cmd/cluster-driver-registrar/main.go]: nit: log level 4 for "tracing" type logs

**Review by davidz627 — COMMENTED** [review]
mostly lgtm, could you add a unit test

File: cmd/cluster-driver-registrar/main.go (hunk 1 above)
Comment (jarrpa) [reply cmd/cluster-driver-registrar/main.go]: Sure.

**Comment by jarrpa** [comment]
@davidz627 Sure, would that just go in `pkg/connection/connection_test.go`?

**Comment by jarrpa** [comment]
@davidz627 Also, is there a canonical / preferred method for running unit tests locally for this project?

**Comment by msau42** [comment]
make test

**Comment by jarrpa** [comment]
PR updated with new commits to address review comments. @davidz627 PTAL

File: cmd/cluster-driver-registrar/main.go [hunk 2]
```diff
@@ -84,17 +79,40 @@ func main() {
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name: csiDriverName,
+		},
+		Spec: k8scsi.CSIDriverSpec{
+			AttachRequired:        &k8sAttachmentRequired,
+			PodInfoOnMountVersion: k8sPodInfoOnMountVersion,
+		},
+	}
+
+	glog.V(2).Infof("CSIDriver object: %v", *csiDriver)
```
Comment (msau42) [inline cmd/cluster-driver-registrar/main.go]: Should this be "%+v"?

**Review by msau42 — COMMENTED** [review]
lgtm, just one nit

File: cmd/cluster-driver-registrar/main.go (hunk 2 above)
Comment (jarrpa) [reply cmd/cluster-driver-registrar/main.go]: Sure thing, updated.

**Comment by msau42** [comment]
lgtm, please squash

**Comment by jarrpa** [comment]
Squashed most of the patched, left the first one on its own since it's just minor stuff.

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**